cpp/src/slice2cs/Gen.cpp

```diff
@@ -157,110 +157,191 @@ Slice::CsVisitor::~CsVisitor()
 }
 
 void
-Slice::CsVisitor::writeMarshalParams(const OperationPtr& op,
-                                     const MemberList& requiredParams,
-                                     const MemberList& taggedParams,
-                                     bool isReturnValue,
-                                     const string& customStream,
-                                     const string& obj)
+Slice::CsVisitor::writeMarshal(const OperationPtr& operation, bool returnType)
 {
-    const string stream = customStream.empty() ? "ostr" : customStream;
-    string ns = getNamespace(op->interface());
+    const string stream = "ostr";
+    string ns = getNamespace(operation->interface());
+
+    MemberList members = returnType ? operation->returnValues() : operation->parameters();
+    auto [requiredMembers, taggedMembers] = getSortedMembers(members);
 
     int bitSequenceIndex = -1;
     size_t bitSequenceSize = 0;
-    if (requiredParams.size() > 0)
+    if (requiredMembers.size() > 0)
```

Comment:
we can put the code below in a single line

```
bitSequenceSize = returnType ? operation->returnBitSequenceSize() : operation->paramsBitSequenceSize();
```

Reply:
Fixed as suggested.